building 4.1.0 docs fails in autosummary:  module 'mpi4py.MPI' has no attribute 'registry'

I'm preparing the debian package for mpi4py 4.1.0. The module builds (PEP517 build using pyproject.toml).
But the doc build fails:
```
$ cd docs/source
$ PYTHONPATH=../../.pybuild/cpython3_3.13/build/ make html
Running Sphinx v8.1.3
loading translations [en]... done
[autosummary] generating autosummary for: changes.rst, citation.rst, develop.rst, guidelines.rst, index.rst, install.rst, intro.rst, license.rst, mpi4py.MPI.rst, mpi4py.bench.rst, ..., mpi4py.run.rst, mpi4py.typing.rst, mpi4py.util.dtlib.rst, mpi4py.util.pkl5.rst, mpi4py.util.pool.rst, mpi4py.util.rst, mpi4py.util.sync.rst, overview.rst, reference.rst, tutorial.rst

Extension error (sphinx.ext.autosummary):
Handler <function process_generate_options at 0x7f61543e79c0> for event 'builder-inited' threw an exception (exception: module 'mpi4py.MPI' has no attribute 'registry')
make: *** [Makefile:20: html] Error 2
```

This is with sphinx  8.1.3, building against OpenMPI 5.0.7.

--- a/.github/workflows/ci-abi.yml
+++ b/.github/workflows/ci-abi.yml
@@ -395,6 +395,10 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v5
 
+    - name: Set hostname
+      run:  sudo hostname localhost
+      if:   runner.os == 'macOS'
+
     - name: Test mpi4py cmdline (mpi-library)
       run:  mpiexec -n 1 python -m mpi4py --mpi-library
     - name: Test mpi4py cmdline (mpi-std-version)

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -270,10 +270,10 @@ def _setup_autodoc(app):
 
     def get_documenter(*args, **kwargs):
         if hasattr(autosummary, "_get_documenter"):
-            obj, args = args[0], args[1:]
+            obj = args[0]  # signature is (obj, parent, *, registry)
             get_documenter = autosummary._get_documenter
         else:
-            obj, args = args[1], (args[0], *args[2:])
+            obj = args[1]  # signature is (app, obj, parent)
             get_documenter = autosummary.get_documenter
         if isinstance(obj, type) and issubclass(obj, BaseException):
             caller = sys._getframe().f_back.f_code.co_name
@@ -281,7 +281,7 @@ def _setup_autodoc(app):
                 if obj.__module__ == "mpi4py.MPI":
                     if obj.__name__ == "Exception":
                         return ExceptionDocumenterCustom
-        return get_documenter(obj, *args, **kwargs)
+        return get_documenter(*args, **kwargs)
 
     from sphinx.ext.autosummary import generate
 

